### PR TITLE
Chasing yaks down dark alleyways

### PR DESCRIPTION
**Pull request by yaleman**
* adds a bunch more test coverage
* chased down some more strings

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

### CODE REVIEW

File: libs/client/src/lib.rs [hunk 1]
```diff
@@ -1647,6 +1673,7 @@ impl KanidmClient {
             .await
     }
 
+    // TODO: add test coverage
```
Comment (Firstyear) [inline libs/client/src/lib.rs:1676]: I thought we did have coverage of this?

File: libs/client/src/service_account.rs [hunk 2]
```diff
@@ -158,6 +161,7 @@ impl KanidmClient {
             .await
     }
 
+    // TODO: test coverage for this, but there's a weird issue with ACPs on apply
```
Comment (Firstyear) [inline libs/client/src/service_account.rs:164]: This function will probably will go away in future
it was for a migration a few versions back and shouldn't be needed now.

File: libs/client/src/sync_account.rs [hunk 3]
```diff
@@ -78,6 +78,7 @@ impl KanidmClient {
             .await
     }
 
+    /// Creates a sync token for a given username
```
Comment (Firstyear) [inline libs/client/src/sync_account.rs:81]: For a given sync-account, not username.

File: server/lib/src/macros.rs [hunk 4]
```diff
@@ -119,16 +119,22 @@ macro_rules! run_create_test {
     }};
 }
 
-// #[macro_export]
 #[cfg(test)]
+/// Runs a test with preloaded entries, then modifies based on a filter/list, then runs a given check
```
Comment (Firstyear) [inline server/lib/src/macros.rs:123]: Funny, as I'm about to try to kill these macros. They're blocking us from removing async-std.

File: server/lib/src/plugins/attrunique.rs [hunk 5]
```diff
@@ -557,24 +550,15 @@ mod tests {
...
     fn test_pre_modify_name_unique() {
-        let ea: Entry<EntryInit, EntryNew> = Entry::unsafe_from_entry_str(
-            r#"{
-            "attrs": {
-                "class": ["group"],
-                "name": ["testgroup_a"],
-                "description": ["testgroup"]
-            }
-        }"#,
+        let ea: Entry<EntryInit,EntryNew> = entry_init!(
```
Comment (Firstyear) [inline server/lib/src/plugins/attrunique.rs:553]: :O yeeeeessss lets go, kill unsafe-from-entry-str

File: server/lib/src/plugins/refint.rs [hunk 6]
```diff
@@ -491,28 +492,47 @@ mod tests {
...
+            (Attribute::Class, EntryClass::Group.to_value()),
+            (Attribute::Name, Value::new_iname("testgroup_b")),
+            (Attribute::Description, Value::new_utf8s("testgroup")),
+            (
+                Attribute::Member,
+                Value::Refer(Uuid::parse_str(TEST_TESTGROUP_A_UUID).unwrap())
+            )
         );
 
+        // let ea: Entry<EntryInit, EntryNew> = Entry::unsafe_from_entry_str(
```
Comment (Firstyear) [inline server/lib/src/plugins/refint.rs:514]: Just delete these?

File: server/testkit/tests/proto_v1_test.rs [hunk 7]
```diff
@@ -1457,24 +1464,134 @@ async fn test_server_api_token_lifecycle(rsclient: KanidmClient) {
...
+    assert!(rsclient
+        .idm_service_account_delete(test_service_account_username)
+        .await
+        .is_ok());
+
+
+
+
+    // let's create one and just yolo it into a person
+    // TODO: Turns out this doesn't work because admin doesn't have the right perms to remove `jws_es256_private_key` from the account?
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply server/testkit/tests/proto_v1_test.rs:1569]: I'm not sure here, it was a temporary fn so we may be able to remove it now.

File: server/testkit/tests/proto_v1_test.rs [hunk 8]
```diff
@@ -1793,3 +1922,21 @@ async fn test_server_user_auth_privileged_shortcut(rsclient: KanidmClient) {
         }
     }
 }
+
+// wanna test how long it takes for testkit to start up? here's your biz.
+// turns out  as of 2023-10-11 on my M2 Max, it's about 1.0 seconds per iteration
```
Comment (Firstyear) [inline server/testkit/tests/proto_v1_test.rs:1927]: That's surprisingly fast contrast to other IDM's I've worked on

File: tools/iam_migrations/ldap/Cargo.toml [hunk 9]
```diff
@@ -40,3 +40,6 @@ kanidm_utils_users = { workspace = true }
 [build-dependencies]
 clap = { workspace = true, features = ["derive"] }
 clap_complete = { workspace = true }
+
+[dev-dependencies]
+tracing-forest = "0.1.6"
```
Comment (Firstyear) [inline tools/iam_migrations/ldap/Cargo.toml:45]: Shouldn't it be sketching?

File: libs/client/src/lib.rs (hunk 1 above)
Comment (yaleman) [reply libs/client/src/lib.rs:1676]: Coverage doesn't find it in testing and I can't find references to it in the code? 🤔

File: server/lib/src/plugins/refint.rs (hunk 6 above)
Comment (yaleman) [reply server/lib/src/plugins/refint.rs:514]: oh yeah, testing artifact. will be gone.

File: server/testkit/tests/proto_v1_test.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (yaleman) [reply server/testkit/tests/proto_v1_test.rs:1569]: It's used in the [CLI commands](https://github.com/yaleman/kanidm/blob/b169f7a7e8fba4b06461a7a9c9fe4f12537e55d5/tools/cli/src/cli/serviceaccount.rs#L499), so we'd have to remove that too...

File: tools/iam_migrations/ldap/Cargo.toml (hunk 9 above)
Comment (yaleman) [reply tools/iam_migrations/ldap/Cargo.toml:45]: yep, changed. (to the sketching version which makes the exact same call 🤣 )

File: server/testkit/tests/proto_v1_test.rs (hunk 8 above)
Comment (yaleman) [reply server/testkit/tests/proto_v1_test.rs:1927]: Yeah, I was trying to work out how much effort I should put into trying to optimise it while adding more invocations of it ^_^ It's probably worth investigating at some point, but for now it's a known-known.

Comment (Firstyear) [reply server/testkit/tests/proto_v1_test.rs:1927]: Generally any performance improvement here will probably have a benefit all over the server, so I likely need to do some optimisation at some point.

File: libs/client/src/lib.rs (hunk 1 above)
Comment (Firstyear) [reply libs/client/src/lib.rs:1676]: Maybe it's not tested then 🤔

File: server/testkit/tests/proto_v1_test.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Firstyear) [reply server/testkit/tests/proto_v1_test.rs:1569]: Yeah, I think we may not need that CLI command anymore either. Maybe we make the CLI command as deprecated and get ready to remove it?

[the remark this answers is not in the document]
Comment (yaleman) [reply server/testkit/tests/proto_v1_test.rs:1569]: Done.

File: libs/client/src/lib.rs (hunk 1 above)
Comment (yaleman) [reply libs/client/src/lib.rs:1676]: Future-us problem. 😄

**Review by Firstyear — APPROVED** [review]

**Pull request merged**